funsor/torch.py

```diff
@@ -352,6 +355,22 @@ def eager_binary_tensor_tensor(op, lhs, rhs):
     return Tensor(data, inputs, dtype)
 
 
+@eager.register(Contract, Tensor, Tensor, frozenset)
+@contractor
+def eager_contract(lhs, rhs, reduced_vars):
+    inputs = OrderedDict((k, d) for t in (lhs, rhs)
+                         for k, d in t.inputs.items() if k not in reduced_vars)
+    data = opt_einsum.contract(lhs.data, list(lhs.inputs),
+                               rhs.data, list(rhs.inputs),
+                               list(inputs), backend="torch")
+    return Tensor(data, inputs, rhs.dtype)
+
+
+@optimize.register(Contract, Tensor, Tensor, frozenset)
```

Comment:
@eb8680 I believe this accomplishes what was previously accomplished via `ATOMS[1:]`.